pkg/cluster/service.go

```diff
@@ -0,0 +1,129 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/redhat-developer/app-services-cli/pkg/api/kas"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster/kafka"
+	"github.com/redhat-developer/app-services-cli/pkg/cluster/serviceregistry"
+	"github.com/redhat-developer/app-services-cli/pkg/kafka/kafkaerr"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	registryPkg "github.com/redhat-developer/app-services-cli/pkg/serviceregistry"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+)
+
+type Service interface {
+	ResourceExists(ctx context.Context, c *KubernetesCluster, namespace string, serviceName string, opts Options) error
+	CreateResource(ctx context.Context, c *KubernetesCluster, namespace string, serviceID string, opts Options) error
+}
+
+type Kafka struct{}
+
+func (k *Kafka) ResourceExists(ctx context.Context, c *KubernetesCluster, namespace string, serviceName string, opts Options) error {
+
+	path := kafka.GetKafkaConnectionsAPIURL(namespace)
+
+	err := makeRequest(ctx, c.clientset, path, serviceName, opts.Localizer)
+
+	return err
+
+}
+
+func (k *Kafka) CreateResource(ctx context.Context, c *KubernetesCluster, namespace string, serviceID string, opts Options) error {
+
+	api := opts.Connection.API()
+
+	path := kafka.GetKafkaConnectionsAPIURL(namespace)
+
+	kafkaInstance, _, err := api.Kafka().GetKafkaById(ctx, serviceID).Execute()
```

Comment:
My take is that we should validate service id before - once we creating resource we should not print out such breaking validation errors.